presto-main/src/main/java/io/prestosql/sql/planner/ExpressionInterpreter.java

```diff
@@ -711,6 +711,14 @@ protected Object visitComparisonExpression(ComparisonExpression node, Object con
             }
 
             Object right = process(node.getRight(), context);
+            if (right == null && operator != ComparisonExpression.Operator.IS_DISTINCT_FROM) {
+                return null;
```

Comment:
You could return `left IS NOT NULL` here.
Same for the `if (left == null && operator != ComparisonExpression.Operator.IS_DISTINCT_FROM)` case above.

Reply:
Whops, this is `!= IS_DISTINCT_FROM` case.
So maybe add `if (right==null && operator == IS_DISTINCT_FROM=)` special case where you return `left IS NOT NULL`.